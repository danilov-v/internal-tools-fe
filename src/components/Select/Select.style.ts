import styled from 'styled-components';
import { Text } from 'components/layout/Text';

export const Container = styled.div`
  margin-right: -4px;
`;

export const Option = styled.option`
  color: ${({ theme }) => theme.darkGray};
`;

export const StyledSelect = styled.select`
  background: none;
  border: none;
  color: ${({ theme }) => theme.darkGray};
  text-align-last: right;
  width: 100%;
`;

export const ValidationError = styled(Text).attrs(() => ({
  variant: 'error',
}))`
  font-size: 12px;
`;
