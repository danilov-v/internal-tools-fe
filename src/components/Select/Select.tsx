import React from 'react';

import { Container, Option, StyledSelect } from './Select.style';

const Select: React.FC<SelectProps> = ({
  id,
  label,
  name,
  onChange,
  options = [],
  placeholder,
  value = '',
}) => (
  <Container>
    {label && <label htmlFor={id}>{label}</label>}
    <StyledSelect
      name={name}
      onChange={onChange}
      placeholder={placeholder}
      value={value}
    >
      {options.map(option => (
        <Option key={option.value} value={option.value}>
          {option.name}
        </Option>
      ))}
    </StyledSelect>
  </Container>
);

type OptionType = {
  name: string;
  value: string | number;
};

type SelectProps = {
  id?: string;
  label?: string;
  name?: string;
  onChange: (e: React.FormEvent<HTMLSelectElement>) => void;
  options?: Array<OptionType>;
  placeholder?: string;
  value?: string | number;
};

export { Select };
