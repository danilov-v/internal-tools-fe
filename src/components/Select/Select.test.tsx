import React from 'react';
import { render, RenderResult } from '@testing-library/react';

import 'jest-styled-components';

import { Select } from './Select';

const getComponent = (props = {}): RenderResult => {
  const parsedProps = {
    onChange: jest.fn(),
    ...props,
  };

  return render(<Select {...parsedProps} />);
};

describe('Select component', () => {
  it('matches snapshot with only required props', () => {
    expect(getComponent().baseElement).toMatchSnapshot();
  });

  it('matches snapshot with additional props', () => {
    const primarySelect = getComponent({
      label: 'center',
      name: 'name',
      options: [
        { name: 'option 1', value: 'option-1' },
        { name: 'option 2', value: 'option-2' },
      ],
      placeholder: 'placeholder',
    });

    expect(primarySelect.baseElement).toMatchSnapshot();
  });
});
