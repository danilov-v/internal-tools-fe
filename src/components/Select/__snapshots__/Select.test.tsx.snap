// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select component matches snapshot with additional props 1`] = `
<body>
  .c0 {
  margin-right: -4px;
}

.c1 {
  background: none;
  border: none;
  text-align-last: right;
  width: 100%;
}

<div>
    <div
      class="c0"
    >
      <label>
        center
      </label>
      <select
        class="c1"
        name="name"
        placeholder="placeholder"
      >
        <option
          class=""
          value="option-1"
        >
          option 1
        </option>
        <option
          class=""
          value="option-2"
        >
          option 2
        </option>
        <option
          class=""
          disabled=""
          value=""
        >
          Не выбрано
        </option>
      </select>
    </div>
  </div>
</body>
`;

exports[`Select component matches snapshot with only required props 1`] = `
.c0 {
  margin-right: -4px;
}

.c1 {
  background: none;
  border: none;
  text-align-last: right;
  width: 100%;
}

<body>
  <div>
    <div
      class="c0"
    >
      <select
        class="c1"
      >
        <option
          class=""
          disabled=""
          value=""
        >
          Не выбрано
        </option>
      </select>
    </div>
  </div>
</body>
`;
